File: iOS9/UIStackViewDemo/UIStackViewDemo/ViewController.m
//
//  ViewController.m
//  UIStackViewDemo
//
//  Created by Xiangjian Meng on 15/10/15.
//  Copyright © 2015年 mengxiangjian. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIStackView *peopleStackView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}
- (IBAction)changeStackViewAlignment:(UISegmentedControl *)sender {
    
    switch (sender.selectedSegmentIndex)
    {
        case 0:
        {
            self.peopleStackView.alignment = UIStackViewAlignmentFill;
        }
            break;
        case 1:
        {
            self.peopleStackView.alignment = UIStackViewAlignmentTop;
        }
            break;
        case 2:
        {
            self.peopleStackView.alignment = UIStackViewAlignmentCenter;
        }
            break;
        case 3:
        {
            self.peopleStackView.alignment = UIStackViewAlignmentBottom;
        }
            break;
            
        default:
            break;
    }
    
}
- (IBAction)changeStackViewDistribution:(UISegmentedControl *)sender
{
    switch (sender.selectedSegmentIndex)
    {
        case 0:
        {
            self.peopleStackView.distribution = UIStackViewDistributionFill;
        }
            break;
        case 1:
        {
            self.peopleStackView.distribution = UIStackViewDistributionFillEqually;
        }
            break;
        case 2:
        {
            self.peopleStackView.distribution = UIStackViewDistributionFillProportionally;
        }
            break;
        case 3:
        {
            self.peopleStackView.distribution = UIStackViewDistributionEqualSpacing;
        }
            break;
        case 4:
        {
            self.peopleStackView.distribution = UIStackViewDistributionEqualCentering;
        }
            break;
            
        default:
            break;
    }

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOS8/UserNotification/UserNotification/ViewController.swift
//
//  ViewController.swift
//  UserNotification
//
//  Created by Xiangjian Meng on 14-10-15.
//  Copyright (c) 2014年 cn.com.modernmedia. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBAction func sendNotification()
    {
        let localNoti = UILocalNotification();
        localNoti.timeZone = NSTimeZone.defaultTimeZone()
        localNoti.alertBody = "是否打开应用？";
        localNoti.fireDate = NSDate(timeIntervalSinceNow: 5);
        localNoti.category = "alert";
        UIApplication.sharedApplication().scheduleLocalNotification(localNoti);
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let category = UIMutableUserNotificationCategory();
        category.identifier = "alert";
        
        let enterAction = UIMutableUserNotificationAction();
        enterAction.identifier = "enter";
        enterAction.activationMode = .Foreground; // 前台处理
        enterAction.title = "进入";
        
        let rejectAction = UIMutableUserNotificationAction();
        rejectAction.identifier = "reject";
        rejectAction.activationMode = .Background; // 后台处理
        rejectAction.authenticationRequired = true; // background mode 最好需要一下密码或者指纹验证，Foreground设置此属性没有用。
        rejectAction.title = "拒绝";
        
        category.setActions([enterAction,rejectAction], forContext: UIUserNotificationActionContext.Minimal);
        
        let settings = UIUserNotificationSettings(forTypes: (.Alert | .Badge | .Sound), categories: NSSet(object: category));
        UIApplication.sharedApplication().registerUserNotificationSettings(settings);
        
        
    }
    
    func showEnterAlert()
    {
        let avc = UIAlertController(title: "进来了！", message: "终于进来了！", preferredStyle: UIAlertControllerStyle.Alert);
        let action = UIAlertAction(title: "知道了", style: UIAlertActionStyle.Default, handler: {_ in
            // 这里是action触发的动作。不用像以前alertView实现代理方法了。
        });
        avc.addAction(action);
        self.presentViewController(avc, animated: true, completion: nil);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

